### PR TITLE
NFP for HAMT Enhancements Part 1

### PR DESCRIPTION
**Pull request by jasondegraw**
Pull request overview
---------------------
This is the NFP for the first of a set of enhancements to the HAMT model to add capabilities. See the document for further justification.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: design/FY2019/NFP-HAMT-Enhancements-Pt1.md [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+## I/O Reference ##
+The I/O Reference will be modified to describe the additional moisture source inputs. A moisture source object is attached to a particular layer in a manner similar to the way that heat sources are specified. Two options are being considered.
+
+### Modify Existing Object ###
+
+The first option modifies the existing internal sources object to include the needed moisture-related parameters. This is the preferred approach as it makes fewer changes to the IDD.
+
+```
+Construction:InternalSources,
```
Comment (mjwitte) [inline design/FY2019/NFP-HAMT-Enhancements-Pt1.md]: Questions:
1. Is it feasible for HAMT to have more than one moisture source in a given construction?
2. Would more than one moisture source be useful?
3. Does HAMT currently support internal heat sources?
4. Could HAMT support both a moisture source and a heat source in the same construction? (I guess the proposed IDD object says Yes to this.)

File: design/FY2019/NFP-HAMT-Enhancements-Pt1.md [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+       \type object-list
+       \object-list MaterialName
+```
+
+### Modify Existing Object, Add Two New Objects ###
+
+This option modifies the existing sources object to move the source information (both thermal and moisture) out into separate objects.
+
+```
+Construction:InternalHeatSource,
```
Comment (mjwitte) [inline design/FY2019/NFP-HAMT-Enhancements-Pt1.md]: I lean toward this format, but suggest ConstructionProperty:InternalHeatSource, have it reference the construction name and material layer number and then the existing Construction:InternalSource transitions to plain old Construction plus the new ConstructionProperty:InternalHeatSource. 

OR if you don't want to touch Construction:InternalSource, how about two new object just for HAMT that are parallel to the existing HAMT objects:
MaterialProperty:HeatAndMoistureTransfer:HeatSource
MaterialProperty:HeatAndMoistureTransfer:MoistureSource

File: design/FY2019/NFP-HAMT-Enhancements-Pt1.md (hunk 1 above)
Comment (jasondegraw) [reply design/FY2019/NFP-HAMT-Enhancements-Pt1.md]: @mjwitte Thanks for the review, here are some answers:

1. I believe so, yes. I don't think there are any technical complications that would prevent it from being done.
2. Possibly, that's a point of active discussion right now. For situations where moisture accumulates near the surface of a material we might want to be able to have separate sources to account for which direction the air is going.
3. I think so, but I'll need to check to be sure.
4. Same as for number 3. The IDD changes were mainly driven by the possibility of multiple sources, but I'd like to be flexible in any case.

File: design/FY2019/NFP-HAMT-Enhancements-Pt1.md (hunk 2 above)
Comment (jasondegraw) [reply design/FY2019/NFP-HAMT-Enhancements-Pt1.md]: OK, I'll make some edits with your naming and go with the "separate objects" plan.

**Comment by Myoldmopar** [comment]
@jasondegraw just to tie up this on my end, I don't have anything to add from what @mjwitte already commented.  This will be a nice addition.

**Comment by jasondegraw** [comment]
@mjwitte @Myoldmopar So can we merge this then?

**Comment by Myoldmopar** [comment]
It's more typical to just leave it open and have the code, idd, idf, tests, and docs all go into develop at the same time.  Is it anticipated that this NFP will build into the full implementation?

**Comment by jasondegraw** [comment]
I'm not with you on it being typical, but in this case there's already a PR (or two) and a branch for the code. So I'd prefer to merge this PR and revive the (closed) PR for the code.

**Comment by Myoldmopar** [comment]
Understood.  Merging this in.

**Pull request merged**